fix: eslint config updates

jharvey10 opened the pull request:
- remove dependency on carbon eslint configs
- add new rule for requiring comments on eslint disable directives
- enforces no warnings during CI linting
- Refactor some class components to functional components
- Fix eslint errors that were being hidden
- Remove top-level react imports in jsx files
- add muted prop to mdx video to satisfy a11y requirements

jharvey10 committed:
fix: eslint config updates

- remove dependency on carbon eslint configs
- add new rule for requiring comments on eslint disable directives
- enforces no warnings during CI linting
- Refactor some class components to functional components
- Fix eslint errors that were being hidden
- Remove top-level react imports in jsx files
- add muted prop to mdx video to satisfy a11y requirements

jharvey10 committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into fix/eslint-config

jdharvey-ibm commented on services/web-app/components/catalog-multislect-filter/catalog-multiselect-filter.js:
```diff
@@ -105,7 +104,7 @@ const CatalogMultiselectFilter = ({
       <PopoverContent className={styles.content} ref={contentRef}>
         {/* div wrapper addeed to receive focus */}
-        {/* eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex */}
+        {/* eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex -- TODO: add role */}
```
Worked with Andrea on this and we concluded that we need guidance from the system squad or accessibility team on what role this container should have (as well as its children)

jdharvey-ibm replied:
Opened #666 to track

jharvey10 committed:
deps: update node modules

francinelucca reviewed (dismissed):
LGTM

jharvey10 committed:
fix(web-app): import, type, and functional component usage

- Convert class components to functional components
- Add 'true' arg to fix carbon type-style issues
- Remove react default imports
- Remove usages of @carbon/icons-react

jharvey10 committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into fix/eslint-config

jharvey10 committed:
fix(web-app): linter errors

jharvey10 committed:
fix(web-app): remove remaining icons-react imports

jharvey10 committed:
ci: add top-level clean script

jharvey10 committed:
refactor(web-app): clean up code smells from nav tree
